smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/middlewares/OperationInputBodyMiddleware.kt

```diff
@@ -25,43 +40,155 @@ class OperationInputBodyMiddleware(
     override val position = MiddlewarePosition.AFTER
 
     override fun render(
+        ctx: ProtocolGenerator.GenerationContext,
         writer: SwiftWriter,
         op: OperationShape,
         operationStackName: String,
     ) {
+        if (!alwaysSendBody && !MiddlewareShapeUtils.hasHttpBody(ctx.model, op)) { return }
+        val writingClosureUtils = WritingClosureUtils(ctx, writer)
+        val documentWritingClosureUtils = DocumentWritingClosureUtils(ctx, writer)
         val inputShape = MiddlewareShapeUtils.inputShape(model, op)
-        val inputShapeName = MiddlewareShapeUtils.inputSymbol(symbolProvider, model, op).name
-        val outputShapeName = MiddlewareShapeUtils.outputSymbol(symbolProvider, model, op).name
-        val xmlName = inputShape.getTrait<XmlNameTrait>()?.value
+        val inputSymbol = symbolProvider.toSymbol(inputShape)
+        val outputSymbol = MiddlewareShapeUtils.outputSymbol(symbolProvider, model, op)
+        val writerSymbol = documentWritingClosureUtils.writerSymbol()
+        var payloadShape = inputShape
+        var keyPath = "\\.self"
+        var payloadWritingClosure = writingClosureUtils.writingClosure(payloadShape)
+        var documentWritingClosure = documentWritingClosureUtils.closure(payloadShape)
+        var isPayloadMember = false
+        var defaultBody = "\"{}\"".takeIf { ctx.service.hasTrait<AwsJson1_0Trait>() || ctx.service.hasTrait<AwsJson1_1Trait>() || ctx.service.hasTrait<RestJson1Trait>() } ?: "nil"
+        var payloadMember = inputShape.members().find { it.hasTrait<HttpPayloadTrait>() }
+        if (payloadMember != null) {
+            payloadShape = ctx.model.expectShape(payloadMember.target)
+            val memberName = ctx.symbolProvider.toMemberName(payloadMember)
+            keyPath = writer.format("\\.\$L", memberName)
+            payloadWritingClosure = writingClosureUtils.writingClosure(payloadMember)
+            documentWritingClosure = documentWritingClosureUtils.closure(payloadMember)
+            isPayloadMember = true
+        }
```

Comment:
In kotlin its best to avoid explicit null checks and instead use a safe operator `?.` combined with `let`

```
payloadMember?.let { member ->
    payloadShape = ctx.model.expectShape(member.target)
    val memberName = ctx.symbolProvider.toMemberName(member)
    keyPath = writer.format("\\.\$L", memberName)
    payloadWritingClosure = writingClosureUtils.writingClosure(member)
    documentWritingClosure = documentWritingClosureUtils.closure(member)
    isPayloadMember = true
}
```

Reply:
Fixed